packages/js/src/settings/components/formik-page-select-field.js

```diff
@@ -0,0 +1,154 @@
+/* eslint-disable complexity */
+import { DocumentAddIcon } from "@heroicons/react/outline";
+import { useCallback, useEffect, useMemo, useState } from "@wordpress/element";
+import { __ } from "@wordpress/i18n";
+import { AutocompleteField, Spinner } from "@yoast/ui-library";
+import classNames from "classnames";
+import { useField } from "formik";
+import { debounce, find, isEmpty, map, values } from "lodash";
+import PropTypes from "prop-types";
+import { ASYNC_ACTION_STATUS } from "../constants";
+import { useDispatchSettings, useSelectSettings } from "../hooks";
+
+/**
+ * @param {JSX.node} children The children.
+ * @param {string} [className] The className.
+ * @returns {JSX.Element} The page select options content decorator component.
+ */
+const PageSelectOptionsContent = ( { children, className = "" } ) => (
+	<div className={ classNames( "yst-flex yst-items-center yst-justify-center yst-gap-2 yst-py-2 yst-px-3", className ) }>
+		{ children }
+	</div>
+);
+
+PageSelectOptionsContent.propTypes = {
+	children: PropTypes.node.isRequired,
+	className: PropTypes.string,
+};
+
+/**
+ * @param {Object} props The props object.
+ * @param {string} props.name The field name.
+ * @param {string} props.id The field id.
+ * @param {string} props.className The className.
+ * @param {string} props.description The description.
+ * @returns {JSX.Element} The page select component.
+ */
+const FormikPageSelectField = ( { name, id, className = "", ...props } ) => {
+	const siteBasicsPolicies = useSelectSettings( "selectPreference", [], "siteBasicsPolicies", {} );
+	const pages = useSelectSettings( "selectPagesWith", [ siteBasicsPolicies ], values( siteBasicsPolicies ) );
+	const { fetchPages } = useDispatchSettings();
+	const [ { value, ...field }, , { setTouched, setValue } ] = useField( { type: "select", name, id, ...props } );
+	const [ status, setStatus ] = useState( ASYNC_ACTION_STATUS.idle );
+	const [ queriedPageIds, setQueriedPageIds ] = useState( [] );
```

Comment:
I think these are not needed anymore since the action should reliably update the `pages` in the store.
So I propose to remove this concept in favor of just looping over the pages. The only thing it seems to do is not show any when you have an error. But you can just check the status for that?

Reply:
It is also used so that if you search in box a you still have the results in box b. Did that not work for you?